### PR TITLE
Make CUDA bhalf conversion code more forward compatible

### PR DESCRIPTION
**Pull request by dalg24**
Introduce macro for NVIDIA GPU Ampere+ architectute (CC >= 8.X) so new architectures don't need to be listed as they come.
Loosely related to #5948, partially addressing https://github.com/kokkos/kokkos/pull/5867#discussion_r1122727975

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp [hunk 1]
```diff
@@ -260,10 +260,16 @@ KOKKOS_INLINE_FUNCTION
 
 /************************** bhalf conversions *********************************/
 // Go in this branch if CUDA version is >= 11.0.0 and less than 11.1.0 or if the
-// architecture is not Ampere
+// architecture is older than Ampere
+#if !defined(KOKKOS_ARCH_KEPLER) && !defined(KOKKOS_ARCH_MAXWELL) && \
+    !defined(KOKKOS_ARCH_PASCAL) && !defined(KOKKOS_ARCH_VOLTA) &&   \
+    !defined(KOKKOS_ARCH_TURING75)
+#define KOKKOS_IMPL_ARCH_NVIDIA_GPU_AMPERE_PLUS
```
Comment (PhilMiller) [inline core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp:267]: Do we have other places in the code where we essentially need to test CUDA compute capability?

This is fine as a one-off internal flag, but we should consider whether something broader would be more helpful down the line.

**Review by PhilMiller — APPROVED** [review]
I'm happy to see this merged in advance of a broader facility

File: core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp (hunk 1 above)
Comment (PhilMiller) [reply core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp:267]: That's being addressed in #5948

**Comment by dalg24** [comment]
Retest this please

**Review by crtrott — APPROVED** [review]

**Comment by crtrott** [comment]
All CUDA Testing passed.

**Review by e10harvey — COMMENTED** [review]
Thank you, @dalg24 !

**Pull request merged**